manifests/component/lookup_table.pp

```diff
@@ -0,0 +1,39 @@
+define rsyslog::component::lookup_table (
+  Integer          $priority,
+  String           $target,
+  String           $confdir,
+  Hash             $lookup_json,
+  String           $lookup_file,
+  Boolean          $reload_on_hup,
+  Optional[String] $format = '<%= $content %>'
+) {
+
+  include rsyslog
+
+  file { "rsyslog::component::lookup_table_json::${title}":
+    path    => "${target}.json",
+    content => inline_template('<%= JSON.pretty_generate @lookup_json %>'),
```

Comment:
Is JSON.pretty_generate always guaranteed to return exact same string with the keys in the same order?